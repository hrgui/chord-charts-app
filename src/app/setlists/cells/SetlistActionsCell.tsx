import * as React from "react";
import SetlistActions from "../SetlistActions";

interface SetlistActionsCellProps {
  data?: any;
}

class SetlistActionsCell extends React.Component<any, any> {
  render() {
    const { data } = this.props;
    return <SetlistActions setlist={data} />;
  }
}

export default SetlistActionsCell;
