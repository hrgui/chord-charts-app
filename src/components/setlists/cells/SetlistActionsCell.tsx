import SetlistActions from "../SetlistActions";

interface SetlistActionsCellProps {
  data?: any;
}

export function SetlistActionsCell({ data }: SetlistActionsCellProps) {
  return <SetlistActions setlist={data} />;
}

export default SetlistActionsCell;
