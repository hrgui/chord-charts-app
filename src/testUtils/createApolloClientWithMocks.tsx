import { ApolloClient } from "@apollo/client";
import { InMemoryCache } from "@apollo/client";

export default function createApolloClientWithMocks(context) {
  return new ApolloClient({
    cache: new InMemoryCache(),
  });
}
