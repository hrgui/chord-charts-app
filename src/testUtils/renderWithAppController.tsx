import React from "react";
import { render, act, wait } from "@testing-library/react";
import { AppController } from "app/core/AppController";

export async function renderWithAppController(
  ui,
  { store, reduxStore, ...appControllerProps }: any = {}
) {
  let el;
  act(() => {
    el = render(
      <AppController
        reduxStore={reduxStore || store}
        testMode
        {...appControllerProps}
      >
        {ui}
      </AppController>
    );
  });
  // AppThemeProvider re-renders twice
  await wait();

  const { rerender: _rerender, ...otherElProps } = el;

  return {
    ...otherElProps,
    rerender: ui => {
      return _rerender(
        <AppController
          reduxStore={reduxStore || store}
          testMode
          {...appControllerProps}
        >
          {ui}
        </AppController>
      );
    }
  };
}
