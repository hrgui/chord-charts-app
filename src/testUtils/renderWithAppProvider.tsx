import React from "react";
import { render, act, RenderResult } from "@testing-library/react";
import { AppProvider } from "app/core/AppProvider";
import { Router as TestRouter } from "react-router-dom";
import { createStore } from "app/store";

export interface RenderWithAppControllerProps {
  store?;
  [name: string]: any;
}

export function renderWithAppController(
  ui,
  {
    store = createStore(),
    gqlMocks = [],
    mockedProviderProps = {},
    ...appControllerProps
  }: RenderWithAppControllerProps = {}
): RenderResult {
  const TestComponentProviderOverrides = {
    Router: appControllerProps.history && TestRouter,
  };

  let el;
  act(() => {
    el = render(
      <AppProvider
        componentProviderOverrides={TestComponentProviderOverrides}
        store={store}
        apolloProviderProps={{ mocks: gqlMocks, ...mockedProviderProps }}
        {...appControllerProps}
      >
        {ui}
      </AppProvider>
    );
  });

  const { rerender: _rerender, ...otherElProps } = el;

  return {
    ...otherElProps,
    rerender: (ui) => {
      return _rerender(
        <AppProvider
          componentProviderOverrides={TestComponentProviderOverrides}
          store={store}
          apolloProviderProps={{ mocks: gqlMocks, ...mockedProviderProps }}
          {...appControllerProps}
        >
          {ui}
        </AppProvider>
      );
    },
  };
}
