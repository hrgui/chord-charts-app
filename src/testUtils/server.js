// src/server.js
import { Server, Model } from "miragejs";
import { buildSchema, graphql } from "graphql";

let graphqlSchema = buildSchema(`
  type Query {
    movies: [Movie]
  }
  type Movie {
    id: ID!
    title: String!
  }
`);

export function makeServer({ environment = "development" } = {}) {
  console.log("i did make the server");
  let server = new Server({
    environment,
    seeds(server) {
      server.create("user", { name: "Bob" });
      server.create("user", { name: "Alice" });
    },

    routes() {
      console.log(" i did call this");

      this.passthrough((req) => {
        console.log(`cypress`, req.url);
        return false;
      });

      this.post("http://localhost:4000/graphql", (schema, request) => {
        let requestJson = JSON.parse(request.requestBody);
        let query = requestJson.query;
        let variables = requestJson.variables;

        let resolver = {
          movies() {
            return schema.db.movies;
          },
        };

        console.log(schema, request);

        return graphql(graphqlSchema, query, resolver, null, variables);
      });
    },
  });

  return server;
}
