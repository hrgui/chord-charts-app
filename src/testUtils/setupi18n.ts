import i18n from "i18next";
import { initReactI18next } from "react-i18next";
const translation = require("../../public/locales/en-US/translation.json");
const setlist = require("../../public/locales/en-US/setlist.json");
const song = require("../../public/locales/en-US/song.json");

i18n.use(initReactI18next).init({
  lng: "en-US",
  fallbackLng: "en-US",
  interpolation: {
    escapeValue: false // not needed for react as it escapes by default
  },
  resources: {
    en: {
      translation,
      setlist,
      song
    }
  }
});
