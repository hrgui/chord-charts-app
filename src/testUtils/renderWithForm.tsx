import { Form } from "lib/form/Form";
import React from "react";
import { renderWithAppController as render } from "./renderWithAppProvider";

interface FormProps {
  initialValues: any;
  onSubmit: any;
  [name: string]: any;
}

export function renderWithForm(
  uiFn,
  formProps: FormProps = { initialValues: {}, onSubmit: () => void 0 }
) {
  const el = render(<Form {...formProps}>{uiFn}</Form>);

  return el;
}
