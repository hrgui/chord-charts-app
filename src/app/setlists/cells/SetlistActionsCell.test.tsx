import * as React from "react";
import { renderWithAppController as render } from "testUtils/renderWithAppProvider";
import SetlistActionsCell from "./SetlistActionsCell";

test("renders setlist actions given a setlist", () => {
  const { container } = render(<SetlistActionsCell data={{}} />);
  expect(container).toMatchSnapshot();
});
