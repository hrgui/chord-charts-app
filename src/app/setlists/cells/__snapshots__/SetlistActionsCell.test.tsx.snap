// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders setlist actions given a setlist 1`] = `
<div>
  <div
    data-theme="light"
  >
    <div
      class="dropdown dropdown-left dropdown-middle"
      role="listbox"
    >
      <label
        tabindex="0"
      >
        <label
          tabindex="0"
        >
          <button
            class="btn"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
              />
            </svg>
          </button>
        </label>
        <ul
          class="dropdown-content menu p-2 shadow bg-base-100 rounded-box w-52"
          tabindex="0"
        >
          <ul>
            <a
              class="flex p-2 items-center hover:bg-base-100"
              href="/setlist/undefined"
            >
              <div
                class="text-white inline-flex min-w-[56px] flex-shrink-0"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11.5 9C10.12 9 9 10.12 9 11.5s1.12 2.5 2.5 2.5 2.5-1.12 2.5-2.5S12.88 9 11.5 9zM20 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm-3.21 14.21l-2.91-2.91c-.69.44-1.51.7-2.39.7C9.01 16 7 13.99 7 11.5S9.01 7 11.5 7 16 9.01 16 11.5c0 .88-.26 1.69-.7 2.39l2.91 2.9-1.42 1.42z"
                  />
                </svg>
              </div>
              <div
                class="flex-[1_1_auto] min-w-0 mt-1 mb-1"
              >
                 
                View
              </div>
            </a>
            <a
              class="flex p-2 items-center hover:bg-base-100"
              href="/setlist/undefined/edit"
            >
              <div
                class="text-white inline-flex min-w-[56px] flex-shrink-0"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                  />
                </svg>
              </div>
              <div
                class="flex-[1_1_auto] min-w-0 mt-1 mb-1"
              >
                 
                Edit
              </div>
            </a>
            <div
              class="flex-[1_1_auto] min-w-0 mt-1 mb-1 flex p-2 items-center hover:bg-base-100 cursor-pointer"
            >
              <div
                class="text-white inline-flex min-w-[56px] flex-shrink-0"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                </svg>
              </div>
              <div
                class="flex-[1_1_auto] min-w-0 mt-1 mb-1"
              >
                 
                Delete
              </div>
               
            </div>
          </ul>
        </ul>
      </label>
      <ul
        class="dropdown-content"
      />
    </div>
  </div>
</div>
`;
